vault.tf

```
@@ -1,21 +1,23 @@
 locals {
   roles = var.vault_backend_path == null ? {} : {
-    "${var.database}-all-privileges" = "ALL PRIVILEGES"
-    "${var.database}-read-only"      = "SELECT"
+    "all-privileges" = "ALL PRIVILEGES"
+    "read-only"      = "SELECT"
   }
 }
 
 resource "vault_database_secret_backend_role" "this" {
   for_each = local.roles
 
-  name    = each.key
+  name    = "${var.database}-${each.key}"
   backend = var.vault_backend_path
   db_name = var.vault_db_connection_name
 
-  creation_statements = [
-    "CREATE USER '{{name}}'@'%' IDENTIFIED BY '{{password}}';",
-    "GRANT ${each.value} ON ${var.database}.* TO '{{name}}'@'%';",
-  ]
+  creation_statements = concat(
+    [
+      "CREATE USER '{{name}}'@'%' IDENTIFIED BY '{{password}}';",
+      "GRANT ${each.value} ON ${var.database}.* TO '{{name}}'@'%';",
+    ], var.roles_extra_statements[each.key] != null ? var.roles_extra_statements[each.key] : [],
```

This looks identical.

```suggestion
    ], coalesce(var.roles_extra_statements[each.key], []),
```